### PR TITLE
Move max_instances_alert_threshold to be per-metrics_provider

### PR DESCRIPTION
**Pull request by EvanKrall**
Seems like max_instances_alert_threshold should be per-metrics_provider (since it's expressed in the same dimensions as setpoint) so here's my attempt at that refactor.

### CODE REVIEW

**Review by drmorr0 — APPROVED** [review]
lgtm, will let @nemacysts chime in before merging

File: paasta_tools/autoscaling/autoscaling_service_lib.py
Comment (nemacysts) [inline paasta_tools/autoscaling/autoscaling_service_lib.py:1]: !8ball deleting this file will be easy

(i don't think any of this is in use atm)

**Review by nemacysts — APPROVED** [review]

File: paasta_tools/autoscaling/autoscaling_service_lib.py
Comment (nemacysts) [reply paasta_tools/autoscaling/autoscaling_service_lib.py:1]: (oh, you didn't actually change things here - GH is just confused diff-wise)

Comment (drmorr0) [reply paasta_tools/autoscaling/autoscaling_service_lib.py:1]: Yea I rebased sorry

**Comment by drmorr0** [comment]
Cherry-picked onto https://github.com/Yelp/paasta/pull/3837, will close this one.

**Pull request closed without merging**